src/Workspaces/Remote/ServiceHub/Shared/Extensions.cs

```diff
@@ -138,25 +138,27 @@ internal static partial class Extensions
 
                     return result;
                 }
-            }
-            catch (Exception ex) when (IsCancelled(ex))
-            {
-                cancellationToken.ThrowIfCancellationRequested();
-                throw;
+                catch (Exception ex) when (IsCancelled(ex, mergedCancellation.Token))
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    throw;
+                }
             }
         }
 
-        private static bool IsCancelled(Exception ex)
+        private static bool IsCancelled(Exception ex, CancellationToken token)
```

Comment:
This method seems poorly named. `IsCanceled` sounds very benign (and the exception type seems unrelated unless it were somehow included in the cancellation check). The fact that the exception crashes the product if it isn't canceled would be totally unexpected given the method name and lack of xml doc comments to help call out otherwise.
Perhaps simply "ThrowIfNotCanceled"?